src/btrfsbalancer.cpp

```diff
@@ -0,0 +1,192 @@
+/****************************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Martin Grimme <martin.grimme@gmail.com>
+** All rights reserved.
+**
+** This file is part of the btrfs-balancer package.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+
+#include "btrfsbalancer.h"
+
+#include <qmath.h>
+#include <QDateTime>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QDebug>
+
+namespace
+{
+
+// the max usages in percents for which to run the balancing operation, in the
+// order of their application
+const QList<int> MAX_USAGE_PERCENTS =
```

Comment:
Hm, maybe just make this a `const int[]` and avoid the `QList` overhead? Or use C++11 initializer lists:

```
const QList<int> MAX_USAGE_PERCENTS({0, 10, 20, 35, 50, 75, 96});
```